### PR TITLE
fix(simple psim): gear bug to update state in simple psim

### PR DESCRIPTION
**Pull request by takayuki5168**
Signed-off-by: Takayuki Murooka <takayuki5168@gmail.com>

## Related Issue(required)

https://github.com/autowarefoundation/autoware.universe/issues/371

## Description(required)

Apply gear to current pose update (previously, only velocity and acceleration was updated)

## Review Procedure(required)

run simple psim

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [commit-guidelines][commit-guidelines]
- [x] Assign PR to reviewer

If you are adding new package following items are required:

- [ ] Documentation with description of the package is available
- [ ] A sample launch file and parameter file are available if the package contains executable nodes

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[commit-guidelines]: https://www.conventionalcommits.org/en/v1.0.0/
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

**Comment by tkimura4** [comment]
Could you fix `ideal_steer_acc_geared` too?

**Comment by takayuki5168** [comment]
@tkimura4 Thanks for the comment. As you may know, Kataoka-san seems to create another PR for ideal one.

**Comment by takayuki5168** [comment]
> Could you fix ideal_steer_acc_geared too?

@tkimura4 I did, and confirmed it works well.

File: simulator/simple_planning_simulator/src/simple_planning_simulator/vehicle_model/sim_model_delay_steer_acc_geared.cpp [hunk 1]
```diff
@@ -58,17 +58,15 @@ void SimModelDelaySteerAccGeared::update(const float64_t & dt)
...
 
   // take velocity limit explicitly
   state_(IDX::VX) = std::max(-vx_lim_, std::min(state_(IDX::VX), vx_lim_));
 
-  state_(IDX::VX) = calcVelocityWithGear(state_, gear_);
-
-  // calc acc directly after gear consideration
-  state_(IDX::ACCX) = (state_(IDX::VX) - prev_vx) / std::max(dt, 1.0e-5);
+  // consider gear
+  // update position and velocity first, and then acceleration is calculated naturally
```
Comment (TakaHoribe) [inline simulator/simple_planning_simulator/src/simple_planning_simulator/vehicle_model/sim_model_delay_steer_acc_geared.cpp:68]: @takayuki5168 I think `a` should be `(v - v_prev) / dt` only when `v=0` is addressed.

**Review by TakaHoribe — APPROVED** [review]
The current update for gear state is like this:
```
if (gear == DRIVE) {
  if (v < 0) {
    v = 0;
    x = x_prev;
    y = y_prev;
    yaw = yaw_prev;
   }
}

a = (v - v_prev) / dt;  // update acceleration anytime.
```
 
But, I suggest taking this way to update the acceleration:

```
if (gear == DRIVE) {
  if (v < 0) {
    v = 0;
    x = x_prev;
    y = y_prev;
    yaw = yaw_prev;
    a = (v - v_prev) / dt;  // updated only when it is in the condition
   }
}

// do not update the acceleration when the above condition does not meet
```

which can be addressed in another PR.

**Pull request merged**